### PR TITLE
Support quaternion to euler conversion

### PR DESCRIPTION
**Pull request by xintongxia**

### CODE REVIEW

**Comment by xintongxia** [comment]
@Pessimistress

**Comment by Pessimistress** [comment]
Add a few test cases?

File: test/utils/tape-equals.js [hunk 1]
```diff
@@ -13,3 +13,14 @@ export function tapeEquals(t, a, b, msg, extra) {
     extra
   });
 }
+
+export function almostEqual(actual, expected) {
```
Comment (Pessimistress) [inline test/utils/tape-equals.js]: Use math.gl's `equal` instead

File: test/src/quaternion.spec.js [hunk 2]
```diff
@@ -83,3 +87,33 @@ test('Quaternion#methods', t => {
 //   t.equals(q1[3], 0.7439232829017486);
 //   t.end();
 // });
+
+test('Quaternion.toEuler', t => {
+  const q = new Quaternion(0, 0, 0.7071067811865475, 0.7071067811865476);
```
Comment (Pessimistress) [inline test/src/quaternion.spec.js]: - Add multiple test cases
- Use something more substantial (no zeros)

File: test/src/euler.spec.js [hunk 3]
```diff
@@ -11,10 +11,12 @@ test('Euler#construct and Array.isArray check', t => {
   t.end();
 });
 
+/* eslint-disable */
 test('Euler.toQuaternion', t => {
   const eulers = [
+    new Euler(90 / 180 * Math.PI, -89 / 180 * Math.PI, - 180 / 180 * Math.PI, Euler.RollPitchYaw),
```
Comment (Pessimistress) [inline test/src/euler.spec.js]: Make a constant `DEGREE_TO_RADIANS`

File: test/src/quaternion.spec.js [hunk 4]
```diff
@@ -83,3 +100,58 @@ test('Quaternion#methods', t => {
 //   t.equals(q1[3], 0.7439232829017486);
 //   t.end();
 // });
+
+test('Quaternion.toEuler', t => {
+  const quaternions = [
```
Comment (Pessimistress) [inline test/src/quaternion.spec.js]: Nit: organize the test cases as an array of `{quaternion, transformMatrix}`

**Comment by ibgreen** [comment]
@xintongxia Nice improvement!

One nit: From a code size perspective it would be desirable not to import Euler from Quaternion - in the way things are done in this PR, any use of quaternion.js will automatically bundle euler.js even if `Euler` is not used (and prevent it from being tree-shaken).

An alternate approach would be to just have an `Euler.fromQuaternion` "static constructor" that simply accesses the elements 0-3 of the quaternion array, without actually importing quaternion.js.

Not a big issue as only some apps use Quaternion, so this can be changed in a future release if too late now.

**Pull request merged**